OpenNefia.Content/UI/Element/FeatWindow.cs

```diff
@@ -0,0 +1,311 @@
+            : base(keyHintXOffset: 64)
+        {
+            _prototypeManager = IoCManager.Resolve<IPrototypeManager>();
+            AssetInventoryIcons = InventoryHelpers.MakeIcon(InventoryIcon.Feat);
+            AssetDecoFeatA = new AssetDrawable(Protos.Asset.DecoFeatA);
+            AssetDecoFeatB = new AssetDrawable(Protos.Asset.DecoFeatB);
+            AssetDecoFeatC = new AssetDrawable(Protos.Asset.DecoFeatC);
+            AssetDecoFeatD = new AssetDrawable(Protos.Asset.DecoFeatD);
+
+            NameTopic = new UiTextTopic(Loc.GetString("Elona.FeatMenu.NameTopic"));
+            DetailTopic = new UiTextTopic(Loc.GetString("Elona.FeatMenu.DetailTopic"));
+            FeatCountText = new UiText(UiFonts.WindowPage);
+
+            TextTitle.Text = Loc.GetString("Elona.FeatMenu.Title");
+            List = new UiPagedList<FeatData>(ItemsPerPage, this, new Vector2i(-55, -3));
+
+            GetFeatsFunc = getFeatsFunc ?? (() => new Dictionary<FeatPrototype, int>());
+            SelectFeatAction = selectFeatAction ?? (feat => { });
+            GetAvailableCountFunc = getAvailableCountFunc ?? (() => 0);
+
+            List.GrabFocus();
+            List.EventOnActivate += List_OnActivate;
+            RefreshData();
+        }
+
+        private void RefreshData(FeatPrototype? lastSelected = null)
+        {
+            List.Clear();
+            
+            var data = new List<FeatCell>();
+            var prototypes = _prototypeManager.EnumeratePrototypes<FeatPrototype>()?.Where(x => x.FeatType == FeatType.Feat)!;
+            var featCount = GetAvailableCountFunc();
+            var gainedFeats = GetFeatsFunc();
+
+            if (featCount > 0)
+            {
+                data.Add(new FeatCell(new FeatData.FeatHeader("Elona.FeatMenu.AvailableHeader")));
+                data.AddRange(prototypes.Select(x =>
+                {
+                    gainedFeats.TryGetValue(x, out var val);
+                    return new FeatCell(new FeatData.Feat(x, val));
+                }));
+                FeatCountText.Text = string.Format(Loc.GetString("Elona.FeatMenu.FeatCount"), featCount);
+            }
+            else
+            {
+                FeatCountText.Text = string.Empty;
+            }
+            data.Add(new FeatCell(new FeatData.FeatHeader("Elona.FeatMenu.GainedHeader")));
+            data.AddRange(gainedFeats
+                .Select(x => new FeatCell(new FeatData.GainedFeat(x.Key, x.Value, x.Key.FeatType == FeatType.Race)))
+                .OrderBy(x => (x.Data as FeatData.GainedFeat)?.IsRace));
+            List.AddRange(data);
+
+            if (lastSelected != null)
+            {
+                try
+                {
+                    var selected = data.First(x => (x.Data as FeatData.GainedFeat)?.Prototype == lastSelected);
+                    var page = (data.IndexOf(selected)) / ItemsPerPage;
+                    List.Muted = true;
```

Comment:
Don't use properties for this, probably make an override in `UiPageList` for muting the sound.

Reply:
changed in b1407d8